### PR TITLE
[SSO] Update IdP Enterprise Admin Views to support OIDC

### PR DESCRIPTION
**Pull request by biyeun**
## Product Description
This adds support for OIDC in the Edit Identity Provider forms accessible through the enterprise console.

Screen Shot 2022-04-22 at 7 39 11 PM

## Technical Summary
https://dimagi-dev.atlassian.net/browse/SAAS-13411

## Feature Flag
Not needed. This code is extremely well tested and users will only see the OIDC updates if an accounting admin has created a IdP that supports OIDC.

### Safety story
Does not impact any active workflows. Code touched is very well tested

### Automated test coverage
Yes

### QA Plan
Not yet needed. QA will test everything once all the SSO work is complete.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by joxl — APPROVED** [review]
Looks good. I'm not familiar enough with Forms usage patterns to know the safety of the edge case below, so I might be missing context there. Just a "general safety" thing I worry about when updating `**kwargs`.

File: corehq/apps/sso/forms.py [hunk 1]
```diff
@@ -766,3 +766,80 @@ def update_identity_provider(self, admin_user):
...
+        required=False,
+    )
+
+    def __init__(self, identity_provider, *args, **kwargs):
+        kwargs['initial'] = {
+            'is_active': identity_provider.is_active,
+            'entity_id': identity_provider.entity_id,
+            'client_id': identity_provider.client_id,
+            'client_secret': identity_provider.client_secret,
+        }
```
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/sso/forms.py:787]: Another option with more flexibility (provide defaults for, but do not overwrite passed in `'initial'` values) would be to use `setdefault` on all of them
```suggestion
        initial = kwargs.setdefault('initial', {})
        initial.setdefault('is_active', identity_provider.is_active)
        initial.setdefault('entity_id': identity_provider.entity_id)
        initial.setdefault('client_id': identity_provider.client_id)
        initial.setdefault('client_secret': identity_provider.client_secret)
```

**Review by millerdev — APPROVED** [review]

**Review by gherceg — COMMENTED** [review]
Just some minor comments

File: corehq/apps/sso/forms.py [hunk 2]
```diff
@@ -241,24 +255,21 @@ def __init__(self, identity_provider, show_help_block=True, *args, **kwargs):
...
-        help_text = _(
-            'This is a high security feature  that ensures Assertions are '
-            'fully encrypted. This feature requires a Premium Azure AD '
-            'subscription. '
-        )
+        if self.idp.idp_type == IdentityProviderType.AZURE_AD:
+            help_text = _(
+                'This is a high security feature  that ensures Assertions are '
+                'fully encrypted. This feature requires a Premium Azure AD '
+                'subscription. '
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/sso/forms.py:266]: Also @joxl pointed out that 
"This is a high security feature  that ensures Assertions are " has a double space between "feature" and "that"

Comment (gherceg) [inline corehq/apps/sso/forms.py:266]:
```suggestion
                'This is a high security feature that ensures Assertions are '
                'fully encrypted. This feature requires a Premium Azure AD '
                'subscription.'
```

**Pull request merged**